lib/lifecycle.js

```diff
@@ -106,10 +106,14 @@ class Lifecycle extends EventEmitter {
   }
 
   registerBeforeStart(scope) {
+    deprecate(`beforeStart is deprecated, please use either didReady or willReady instead. 
+    For more you can refer to https://eggjs.org/en/basics/app-start.html`);
```

Comment:
link to https://eggjs.org/en/advanced/loaderUpdate.html is better than `app-start`

Reply:
OK